### DISCUSSION TITLE
How to fix FASTAPI error "TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body"

### DISCUSSION THREAD

**Discussion by arumds** [fastapi/fastapi#6450] [Questions] [answered]

**Comment by avarf** [comment]
Hi, I am facing this issue right now. While the curl command generated by the Swagger is working fine the Swagger UI itself is not working and I am getting the same error. How did you solve this issue?

**Reply by SAMTEC15** [reply]
How were you able to resolve this issue.

**Reply by shimulch** [reply, marked as answer]
From browser, you can't send a body with `GET` or `HEAD` method. Even though curl might work, you can't do that from browsers.

Ref:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD

**Comment by eightlay** [comment]
@avarf, take a look at https://github.com/tiangolo/fastapi/issues/50

**Comment by thecodecoffeelab** [comment]
Method GET cannot have body, it says the error on the message, make the data you are getting in a query string.

**Comment by alvynabranches** [comment]
TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body.
--


I am getting this error on Swagger UI but when I copy and paste the curl command on terminal it gets the results sucessfully. 

Does anyone has solution to it???

**Comment by SAMTEC15** [comment]
Please, the whole process is not clear as am a beginner learning all on my own.

**Reply by pythonweb2** [reply]
@shimulch answered the question above, please take time to read the links that he posted to help understand the difference between the HTTP methods.

**Comment by r-sniper** [comment]
As per the new [RFC specification updated in 2014](https://www.rfc-editor.org/rfc/rfc7231#page-24), 
_"A payload within a GET request message has no defined semantics;
   sending a payload body on a GET request might cause some existing
   implementations to reject the request."_
   
 So GET with payload(body) is not invalid.
Even POSTMAN supports GET with body now.
